systemvm/patches/debian/config/opt/cloud/bin/configure.py

```diff
@@ -543,9 +543,12 @@ def configure_ipsec(self, obj):
             secret.commit()
             file.commit()
             logging.info("Configured vpn %s %s", leftpeer, rightpeer)
-            CsHelper.execute("ipsec reload")
-            if not obj['passive']:
-                CsHelper.execute("sudo nohup ipsec up vpn-%s &" % rightpeer)
+            CsHelper.execute("ipsec rereadsecrets")
+
+        CsHelper.execute("ipsec reload")
+        if not obj['passive']:
+            CsHelper.execute("sudo nohup ipsec down vpn-%s " % rightpeer)
+            CsHelper.execute("sudo nohup ipsec up vpn-%s &" % rightpeer)
```

Comment:
'ipsec down' we running for the safe side. If the connection is by any chance then we it will make it down and new connection will come up with new configuration.

'ipsec down' will not take much time and completes immediately if the tunnel is up or down. Here ipsec down will run first then ipsec up will go into background.
ipsec up takes longer time when the there is problem in the configuration. So in this the agent command will time out. So for this reason we are running ipsec up in back ground. 

Let me know if you any tunnel establishment issues with it.